plugins/TrollManagement/class.trollmanagement.plugin.php

```diff
@@ -469,4 +505,201 @@ private function checkTroll(int $userID, array &$args) {
+            'TrollManagement.PerFingerPrint.Enabled' => c('TrollManagement.PerFingerPrint.Enabled', false),
+            'TrollManagement.PerFingerPrint.MaxUserAccounts' => c('TrollManagement.PerFingerPrint.MaxUserAccounts', 5),
+        ]);
+
+        $sender->Form->setModel($configurationModel);
+
+        // If seeing the form for the first time...
+        if ($sender->Form->authenticatedPostBack() === false) {
+            $sender->Form->setData($configurationModel->Data);
+        } else {
+            $configurationModel->Validation->applyRule('TrollManagement.PerFingerPrint.Enabled', 'Boolean');
+
+            if ($sender->Form->getFormValue('TrollManagement.PerFingerPrint.Enabled')) {
+                //add custom validation rule
+                $configurationModel->Validation->addRule('validatePositiveNumber', 'function:validatePositiveNumber');
+
+                $configurationModel->Validation->applyRule('TrollManagement.PerFingerPrint.MaxUserAccounts', 'Required');
+                $configurationModel->Validation->applyRule('TrollManagement.PerFingerPrint.MaxUserAccounts', 'Integer');
+                $configurationModel->Validation->applyRule(
+                    'TrollManagement.PerFingerPrint.MaxUserAccounts',
+                    'validatePositiveNumber',
+                    sprintf(t('%s must be a positive number.'), t("Maximum user's accounts"))
+                );
+            }
+
+            if ($sender->Form->save()) {
+                $sender->StatusMessage = t('Your changes have been saved.');
+            }
+        }
+
+        $sender->render('configuration', '', 'plugins/TrollManagement');
+    }
+
+    /**
+     * Add a _probable_ justification as to why a user's is on the applicant's list if there are too many user
+     * accounts using the same fingerprint.
+     *
+     * @param Controller $sender
+     * @param array $args
+     */
+    public function base_applicantInfo_handler($sender, $args) {
+        if (c('TrollManagement.PerFingerPrint.Enabled', false)) {
+            $maxSiblingAccounts = c('TrollManagement.PerFingerPrint.MaxUserAccounts');
+            $userFingerprint = $args['User']['Fingerprint'] ?? '';
+            if (!empty($userFingerprint)) {
+                if ($this->checkMaxSharedFingerprintsExceeded($userFingerprint, $maxSiblingAccounts)) {
+                    $sender->EventArguments['ApplicantMeta'][t("Fingerprint issue")] = sprintf(
+                        t("Too many accounts are using the '%s' fingerprint."),
+                        $userFingerprint
+                    );
+                }
+            }
+        }
+    }
+
+    /**
+     * Checks if the maximum amount of accounts using the same fingerprint has been reached.
+     *
+     * @param string $fingerprint
+     * @param int $maxSiblingAccounts
+     * @return bool
+     */
+    private function checkMaxSharedFingerprintsExceeded(string $fingerprint, int $maxSiblingAccounts): bool {
```

Comment:
We added a type to `$fingerprint`, which is great`, but if `null` is also a valid value, we need to specify that in the type declaration.

```suggestion
    private function checkMaxSharedFingerprintsExceeded(?string $fingerprint, int $maxSiblingAccounts): bool {
```